File: webfrontend/htmlauth/Developer/t2s-interface_not tested.ts
P2W_Text = "Text to be spoken";
psubfolder = "folder where your plugin is installed (according to your plugin.cfg)";

import { connectAsync, IClientOptions, IClientPublishOptions } from 'mqtt';
import { v4 as uuidv4 } from 'uuid';

interface TTSResponse {
  file?: string;
  httpinterface?: string;
  corr?: string;
}

export async function t2svoice(
  P2W_Text: string,
  psubfolder: string,
  mqttCreds: {
    brokerhost?: string;
    brokerport?: number;
    brokeruser?: string;
    brokerpass?: string;
  }
): Promise<string> {
  const RESP_TIMEOUT = 12000; // Timeout in milliseconds

  // ---------- Generate Unique Topics ----------
  const client = psubfolder;
  const corr = uuidv4() || Date.now().toString();
  const req_topic = `tts-publish/${client}/${corr}`;
  const resp_topic = `tts-subscribe/${client}/${corr}`;

  // ---------- Prepare Text Payload ----------
  const cleanText = (P2W_Text || '').replace(/\r?\n|\r/g, '');
  if (cleanText === '') {
    return usepico(); // Fallback if no text
  }

  // ---------- Create JSON Payload ----------
  const payload = JSON.stringify({
    text: cleanText,		// Text to converted into voice mp3 file
    nocache: 0,				// 0 = check if Voice file already created and ship from cache 1 =  force TTS to recreate MP3 file
    logging: 1,				// 0 = no TTS logs were passed. 1 = you get all logs from TTS for further proccessing
    mp3files: 0,			// 0 =   1 = 
    client,					// identifyer or requester (plugin installation folder)
    corr,					// unique identifyer for this specific pub request
    reply_to: resp_topic,	// fallback if corr doesn't work
  });

  // ---------- MQTT Connection Setup ----------
  // insert code to retrieve MQTT connection details (eg. from general.json)
  const {
    brokerhost = '127.0.0.1',
    brokerport = 1883,
    brokeruser = '',
    brokerpass = '',
  } = mqttCreds;

  const mqttOptions: IClientOptions = {
    username: brokeruser || undefined,
    password: brokerpass || undefined,
    reconnectPeriod: 0,
  };

  try {
    const mqttClient = await connectAsync(`mqtt://${brokerhost}:${brokerport}`, mqttOptions);

    let reply: TTSResponse | null = null;

    // Subscribe to response topic
    await mqttClient.subscribeAsync('tts-subscribe/#');

    mqttClient.on('message', (topic, message) => {
      try {
        const data = JSON.parse(message.toString());
        const response = data.response ?? data;
        const parsed: TTSResponse = {
          file: response.file,
          httpinterface: response.interfaces?.httpinterface ?? response.httpinterface,
          corr: response.corr ?? response.original?.corr,
        };
        if (parsed.corr === corr) {
          reply = parsed;
        }
      } catch {
        // Ignore malformed messages
      }
    });

    // Publish request
    const publishOptions: IClientPublishOptions = { qos: 0 };
    mqttClient.publish(req_topic, payload, publishOptions);

    // Wait for response with timeout
    const start = Date.now();
    while (!reply && Date.now() - start < RESP_TIMEOUT) {
      await new Promise((resolve) => setTimeout(resolve, 100));
    }

    await mqttClient.endAsync();

    // ---------- Handle Response or Fallback ----------
    if (reply?.file && reply.httpinterface) {
      const url = `${reply.httpinterface}/${reply.file}`;
		// enter your code for further proccessing
    } else {
		// enter your code for fallback
    }
  } catch {
    // Fallback code if MQTT connection fails
  }
}

File: webfrontend/htmlauth/Developer/t2s-interface_not tested.py
import json
import time
import uuid
import paho.mqtt.client as mqtt # just an example

P2W_Text = "Text to be spoken";
psubfolder = "folder where your plugin is installed (according to your plugin.cfg)";

def t2svoice(P2W_Text, psubfolder, mqtt_creds):
    RESP_TIMEOUT = 12  # seconds

    # ---------- Generate Unique Topics ----------
    client = psubfolder
    corr = str(uuid.uuid4())
    req_topic = f"tts-publish/{client}/{corr}"
    resp_topic = f"tts-subscribe/{client}/{corr}"

    # ---------- Prepare Text Payload ----------
    clean_text = (P2W_Text or "").replace('\n', '').replace('\r', '')
    if clean_text == "":
        # add your Fallback if no text

    payload = json.dumps({
        "text": clean_text,
        "nocache": 0,
        "logging": 1,
        "mp3files": 0,
        "client": client,
        "corr": corr,
        "reply_to": resp_topic
    })

    # ---------- MQTT Setup ----------
    host = mqtt_creds.get("brokerhost", "127.0.0.1")
    port = mqtt_creds.get("brokerport", 1883)
    user = mqtt_creds.get("brokeruser", "")
    password = mqtt_creds.get("brokerpass", "")

    reply = {}

    def on_message(client, userdata, msg):
        try:
            data = json.loads(msg.payload.decode())
            response = data.get("response", data)
            if response.get("corr") == corr or response.get("original", {}).get("corr") == corr:
                reply["file"] = response.get("file")
                reply["httpinterface"] = response.get("interfaces", {}).get("httpinterface") or response.get("httpinterface")
        except Exception:
            pass  # Ignore malformed messages

    try:
        # ---------- MQTT Connection Setup ----------
		# insert code to retrieve MQTT connection details (eg. from general.json)
        if user or password:
            client_mqtt.username_pw_set(user, password)
        client_mqtt.on_message = on_message
        client_mqtt.connect(host, port, keepalive=60)
        client_mqtt.loop_start()
        client_mqtt.subscribe("tts-subscribe/#")
        client_mqtt.publish(req_topic, payload)

        # ---------- Wait for Response ----------
        end_time = time.time() + RESP_TIMEOUT
        while not reply and time.time() < end_time:
            time.sleep(0.1)

        client_mqtt.loop_stop()
        client_mqtt.disconnect()

        # ---------- Handle Response or Fallback ----------
        if reply.get("file") and reply.get("httpinterface"):
            url = f"{reply['httpinterface']}/{reply['file']}"
            global full_path_to_mp3
            full_path_to_mp3 = url
            # enter your code for further proccessing
        else:
            # enter your code for fallback
